planning/behavior_path_planner/autoware_behavior_path_static_obstacle_avoidance_module/README.md

```diff
@@ -747,6 +747,59 @@ The longitudinal positions depends on envelope polygon, ego vehicle specificatio
+
+```yaml
+        car:
+          ...
+          lateral_margin:
+            soft_margin: 0.3                            # [m]
+            hard_margin: 0.2                            # [m]
+            hard_margin_for_parked_vehicle: 0.7         # [m]
+```
+
+Basically, this module tries to generate avoidance path in order to keep lateral distance, which is sum of `soft_margin` and `hard_margin`/`hard_margin_for_parked_vehicle`, from avoidance target object.
+
+![fig](./images/path_generation/soft_hard.png)
+
+But if there isn't enough space to keep `soft_margin` distance, this module shortens soft constraint lateral margin. The parameter `soft_margin` is a maximum value of soft constraint, and actual soft margin can be a value between 0.0 and `soft_margin`. On the other hand, this module definitely keeps `hard_margin` or `hard_margin_for_parked_vehicle` depending on the situation. Thus, the minimum value of total lateral margin is `hard_margin`/`hard_margin_for_parked_vehicle`, and the maximum value is the sum of `hard_margin`/`hard_margin_for_parked_vehicle` and `soft_margin`.
+
+Following figure shows the situation where this module shortens lateral soft constraint in order not to drive opposite direction lane when user set a parameter `use_opposite_lane` to `false`.
+
+![fig](./images/path_generation/adjust_margin.png)
+
+This module avoids not only parked vehicle but also non-parked vehicle which stops temporarily for some reason (e.g. waiting for traffic light to change red to green.). Additionally, this module has two types hard margin parameters, `hard_margin` and `hard_margin_for_parked_vehicle` and judges if it's a parked vehicle or not for each vehicles because it takes the risk of vehicle doors opening suddenly and people getting out from parked vehicle into consideration.
+
+Basically, user had better make `hard_margin_for_parked_vehicle` larger than `hard_margin` to prevent collision with doors or people who suddenly get out from vehicle.
+
+On the other hand, this module has only one parameter `soft_margin` for soft lateral margin constraint.
+
+![fig](./images/path_generation/hard_margin.png)
+
+As the hard margin parameters the distance which the user definitely want to keep, they are used in the logic to check whether the ego can pass side of the target object without avoidance maneuver as well.
+
+If the lateral distance is less than `hard_margin`/`hard_margin_for_parked_vehicle` when assuming that the ego follows current lane without avoidance maneuver, this module thinks the ego can pass the side of the object safely and the ego must avoid it. In this case, this module inserts stop point until the avoidance maneuver is allowed to execute so that the ego can avoid the object after approval. (e.g. The ego keeps stopping in front of such a object until operator approves avoidance maneuver if user uses this module in MANUAL mode.)
```

Comment:
I couldn't understand the part "this module thinks the ego can pass the side of the object safely and the ego must avoid it". Is it safe when the lateral distance is less than hard_margin`/`hard_margin_for_parked_vehicle ? In my understanding this module thinks that the ego should pass the side of the object with a certain amount of lateral distance, and therefore  the ego must avoid the (parked) vehicle.

Reply:
@go-sakayori Sorry, I wrote wrong explanation, so I fixed it in https://github.com/autowarefoundation/autoware.universe/pull/7554/commits/9938ae07674a4edff3d5fdbf688888a598c998bd.

WRONG: this module thinks the ego can pass the side of the object safely and the ego must avoid it.
CORRECT: this module thinks the ego can **NOT** pass the side of the object safely and the ego must avoid it.